### PR TITLE
Test upgrading protobuf to 3.7.x

### PR DESCRIPTION
**Pull request by jtattermusch**

### CODE REVIEW

File: grpc.gemspec [hunk 1]
```diff
@@ -29,7 +29,7 @@ Gem::Specification.new do |s|
   s.require_paths = %w( src/ruby/lib src/ruby/bin src/ruby/pb )
   s.platform      = Gem::Platform::RUBY
 
-  s.add_dependency 'google-protobuf', '~> 3.1'
+  s.add_dependency 'google-protobuf', '~> 3.7.2.rc.2'
```
Comment (jtattermusch) [inline grpc.gemspec]: @apolcyn  what's the right way to specify the 3.7.2 prerelease package? 
https://rubygems.org/gems/google-protobuf/versions/3.7.0.rc.2-x86_64-linux

Comment (jtattermusch) [reply grpc.gemspec]: I'm getting this error: https://source.cloud.google.com/results/invocations/aeae23e1-b6c7-412f-8e58-02ac05cced22/targets/github%2Fgrpc%2Faggregate_tests/tests;group=aggregate_tests;test=run_tests_ruby_linux_dbg_native;row=3

**Comment by jtattermusch** [comment]
I updated the PR to the head of 3.7.x branch to retest if there are any more blockers.

**Comment by jtattermusch** [comment]
There still seems to be some ruby problem: https://source.cloud.google.com/results/invocations/16f8a2e4-a395-42e8-8aad-b4eb36ee830b/targets/github%2Fgrpc%2Fruby_linux_dbg_native/tests

File: grpc.gemspec (hunk 1 above)
Comment (apolcyn) [reply grpc.gemspec]: Sorry. Can we depend on `3.7.0.rc.2`? I think that's related to the below error. `0.rc.2` is the version that's published on rubygems.org

**Comment by TeBoring** [comment]
It seems the ```add_file``` is not available.
We added this to our runtime in this release. Seems you haven't updated ruby runtime but used the new protoc.

File: grpc.gemspec (hunk 1 above)
Comment (jtattermusch) [reply grpc.gemspec]: Ah, it was a typo. I actually rolled back that change from the PR because I wasn't sure how to fix. Oddly, it still fails.

**Comment by jtattermusch** [comment]
I regenerated generated ruby files and bumped the ruby dependency to 3.7.0.rc.2 and test results are not looking pretty good.

**Comment by jtattermusch** [comment]
Superseded by https://github.com/grpc/grpc/pull/18329.

**Pull request closed without merging**